.github/ci.sh

```diff
@@ -204,6 +204,32 @@ zip_dist_with_solvers() {
   tar -cvzf "$sname".tar.gz "$sname"
 }
 
+make_source_distribution() {
+    # Ideally one makes source distributions with "git archive", but
+    # "git archive" and submodules do not work together. 
+    #
+    # We get a freshly checked out tree that's had all its submodules
+    # cloned and checked out. Make a tarball that looks like it was
+    # made with "git archive" but contains the submodules.
+    #
+    # Don't use -v with tar as there's some 12000 lines of output and
+    # not much to be gained by collecting it, especially if something
+    # goes wrong.
+    #
+    # Create the file in distrib/ so it doesn't try to include itself.
+    #
+    # FUTURE: instead of using "now" here, we should have a date stamp
+    # for the release.
+    mkdir distrib
+    tar -czf distrib/"$1"-sources.tar.gz \
+        --mtime=now --mode=go+rX --owner=0 --group=0 --sort=name \
+        --exclude=".git" --exclude="distrib" \
+        --transform="s,^./,$1/," \
```

Comment:
Is `--transform` portable? I don't see it mentioned in the `man tar` output on macOS, which leaves me worried that this is a feature that is present in GNU `tar` but not BSD `tar`...